torchrl/envs/transforms/transforms.py

```diff
@@ -4505,8 +4512,30 @@ def __init__(
         if not isinstance(kwargs, CompositeSpec):
             kwargs = CompositeSpec(kwargs)
         self.primers = kwargs
+        if random and default_value:
+            raise ValueError(
+                "Setting random to True and providing a default_value are incompatible."
+            )
+        default_value = (
+            default_value or 0.0
+        )  # if not random and no default value, use 0.0
         self.random = random
+        if isinstance(default_value, dict):
+            primer_keys = {unravel_key(key) for key in self.primers.keys(True, True)}
+            default_value_keys = {unravel_key(key) for key in default_value.keys()}
```

Comment:
Ah very cool option @vmoens !

atm I need to transform it back to dict, because I get the following behaviour for non-tensor data:

```
import torch
from tensordict import TensorDict
default_value = {
    "mykey1": lambda: torch.ones(3),
    "mykey2": lambda: torch.tensor(1, dtype=torch.int64),
}
default_value = TensorDict(default_value, [])
keys = default_value.keys(True, True)
print(keys)
```

output:
```
_TensorDictKeysView([],
    include_nested=True,
    leaves_only=True)

```

Reply:
Are you using the latest nightly?
You can always define your own is_leaf for keys:
```python
default_value.keys(True, True, is_leaf=lambda x: isinstance(x, (torch.Tensor, NonTensorData)))
```

Reply:
I was using the latest commit, yes. But this solution works fine :)